akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -194,6 +197,7 @@ trait ByteBufferSerializer {
    * Produces an object from a `ByteBuffer`, with an optional type-hint;
    * the class should be loaded using ActorSystem.dynamicAccess.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
There is 2 more place where we should declare this. In `Serialization.scala`:
```
def deserializeByteBuffer(buf: ByteBuffer, serializerId: Int, manifest: String): AnyRef 

def serializerFor(clazz: Class[_]): Serializer
```

The other public methods there are alright because they return `Try`.